### PR TITLE
fix(ingest): webhooks and onevent -> rdkafka

### PR DESCRIPTION
**Pull request by oliverb123**
Hack and slash job as part of incident response

### CODE REVIEW

File: plugin-server/src/main/ingestion-queues/batch-processing/each-batch-onevent.ts [hunk 1]
```diff
@@ -59,15 +53,17 @@ export async function eachMessageAppsOnEventHandlers(
     }
 }
 
-export async function eachBatchAppsOnEventHandlers(
-    payload: EachBatchPayload,
-    queue: KafkaJSIngestionConsumer
-): Promise<void> {
+export async function eachBatchAppsOnEventHandlers(payload: Message[], consumer: IngestionConsumer): Promise<void> {
+    const hub = consumer.pluginsServer
+    if (!consumer.consumer?.consumer) {
```
Comment (pauldambra) [inline plugin-server/src/main/ingestion-queues/batch-processing/each-batch-onevent.ts:58]: surely we could fit one more consumer here 

`!consumer.consumer?.consumer`is a great line of code 🤣

File: plugin-server/src/main/ingestion-queues/batch-processing/each-batch-webhooks.ts [hunk 2]
```diff
@@ -99,50 +103,25 @@ export async function eachBatchHandlerHelper(
...
-            resolveOffset(lastOffset)
-            await commitOffsetsIfNecessary()
-
-            // Record that latest messages timestamp, such that we can then, for
-            // instance, alert on if this value is too old.
-            latestOffsetTimestampGauge
-                .labels({ partition: batch.partition, topic: batch.topic, groupId: key })
-                .set(Number.parseInt(lastTimestamp))
-
-            await heartbeat()
```
Comment (pauldambra) [inline plugin-server/src/main/ingestion-queues/batch-processing/each-batch-webhooks.ts:138]: we lost a heartbeat here... idk if that matters?

File: plugin-server/src/main/ingestion-queues/batch-processing/each-batch-webhooks.ts [hunk 3]
```diff
@@ -99,50 +103,25 @@ export async function eachBatchHandlerHelper(
...
-            )
-
-            resolveOffset(lastOffset)
-            await commitOffsetsIfNecessary()
-
-            // Record that latest messages timestamp, such that we can then, for
-            // instance, alert on if this value is too old.
-            latestOffsetTimestampGauge
-                .labels({ partition: batch.partition, topic: batch.topic, groupId: key })
-                .set(Number.parseInt(lastTimestamp))
```
Comment (pauldambra) [inline plugin-server/src/main/ingestion-queues/batch-processing/each-batch-webhooks.ts:136]: do we need to alert on this?

File: plugin-server/src/main/ingestion-queues/on-event-handler-consumer.ts [hunk 4]
```diff
@@ -74,70 +48,59 @@ export const startAsyncWebhooksHandlerConsumer = async ({
...
+            hub.groupTypeManager,
+            hub.organizationManager,
+            hub.postgres
+        )
+    }
+
+    const consumer = new IngestionConsumer(
+        hub,
+        KAFKA_EVENTS_JSON,
+        `${KAFKA_PREFIX}clickhouse-plugin-server-async-onevent`,
```
Comment (pauldambra) [inline plugin-server/src/main/ingestion-queues/on-event-handler-consumer.ts:81]: was async-webhooks on deleted line here https://github.com/PostHog/posthog/pull/26465/files#diff-3e6b93346e13fb7e33bb6f16a2952b8457c58f0495c09c21906c07c6829d4799L79

should this be async-webhook?

**Review by pauldambra — APPROVED** [review]
few questions from scanning code I don't know
but incident 🚢 , will trust you :)

File: plugin-server/src/main/ingestion-queues/on-event-handler-consumer.ts (hunk 4 above)
Comment (oliverb123) [reply plugin-server/src/main/ingestion-queues/on-event-handler-consumer.ts:81]: Yes, thank you

File: plugin-server/src/main/ingestion-queues/batch-processing/each-batch-onevent.ts (hunk 1 above)
Comment (daibhin) [reply plugin-server/src/main/ingestion-queues/batch-processing/each-batch-onevent.ts:58]: image

**Pull request closed without merging**